src/components/DonationForm/Forms/UpgradeToMonthlyForm.vue

```diff
@@ -0,0 +1,165 @@
+								class="wmde-banner-select-group-input"
+								@change="onChange"
+							/>
+							<span class="wmde-banner-select-group-label">
+                                {{ $translate( 'upgrade-to-monthly-yes', { amount: secondPageAmountMonthlyFraction } ) }}
+                            </span>
+						</label>
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<a
+			tabIndex="-1"
+			href="#"
+			class="wmde-banner-form-upgrade-custom t-annual-upgrade-yes-custom"
+			:class="{ 'wmde-banner-form-upgrade-custom--with-error': intervalValidity === Validity.Invalid }"
+			@click.prevent="onGoToChangeOfAmount"
+		>
+			{{ $translate( 'upgrade-to-monthly-link' ) }}
+		</a>
+
+		<span v-if="intervalValidity === Validity.Invalid" class="wmde-banner-select-group-error-message">
+            <span class="wmde-banner-error-icon">
+                {{ $translate( 'upgrade-to-monthly-error' ) }}
+            </span>
+        </span>
+
+		<div class="wmde-banner-form-button-container upgrade-to-yearly-button">
+			<button tabIndex="-1" class="wmde-banner-form-button t-submit-upgrade-to-yearly" type="submit">
+				{{ $translate( 'upgrade-to-monthly-button' ) }}
+			</button>
+		</div>
+
+	</form>
+</template>
+
+<script lang="ts">
+// All form components must have names
+export default {
+	name: 'UpgradeToMonthlyForm'
+};
+</script>
+<script setup lang="ts">
+
+import { computed, inject, ref, watch } from 'vue';
+import { useFormModel } from '@src/components/composables/useFormModel';
+import { Validity } from '@src/utils/FormModel/Validity';
+import ChevronLeftIcon from '@src/components/Icons/ChevronLeftIcon.vue';
+import { Intervals } from '@src/utils/FormItemsBuilder/fields/Intervals';
+import { Currency } from '@src/utils/DynamicContent/formatters/Currency';
+import { Tracker } from '@src/tracking/Tracker';
+import { UpgradeToMonthlyEvent } from '@src/tracking/events/UpgradeToMonthlyEvent';
+import { useFormStepShownEvent } from '@src/components/DonationForm/Forms/useFormStepShownEvent';
+
+interface Props {
+	isCurrent: boolean
+}
+
+const props = defineProps<Props>();
+const emit = defineEmits( [ 'submit', 'previous' ] );
+
+const tracker = inject<Tracker>( 'tracker' );
+const interval = ref<string>( '' );
+const intervalValidity = ref<Validity>( Validity.Unset );
+
+const { numericAmount } = useFormModel();
+const currencyFormatter = inject<Currency>( 'currencyFormatter' );
+const suggestedMonthlyFraction = ref( 0 );
+
+const secondPageAmount = computed( (): string => currencyFormatter.euroAmount( numericAmount.value ) );
+const secondPageAmountMonthlyFraction = ref( '' );
+
+useFormStepShownEvent( 'UpgradeToMonthlyForm', tracker, props );
+
+// dynamic read-only access for the template variables (so they don't update when the user updates the amount selection on the second page)
+watch( numericAmount, () => {
+	if ( !props.isCurrent ) {
+		suggestedMonthlyFraction.value = Math.ceil( numericAmount.value / 10 );
```

Comment:
I can understand the intention of doing it like that, but that would also assume there is almost not sense behind those suggested numbers and they have to be hardcoded all the time (which is not the case so far).
I'd (for now)  suspect that the suggested values will stay kind of deductible.
Maybe we can wait for 1 more test around this feature and see how they might develop it before choosing to hardcode a map like this.

Reply:
Yeah but that'll mean that this form page isn't backwards compatible at that point, editing it will break the archive for this test and we'll have to create a new one. So making it a bit more flexible now is worth it. Plus it's way more explicit.

>there is no relation between the original values and the suggested numbers

There should be no relation between the range and the suggested value. If you look at the WMF banners they don't suggest based on math, and that makes sense because we can then suggest different values for different ranges and not just always 10%.

>choosing to hardcode a map like this

The map shouldn't be hardcoded, it should be passed as a prop in the BannerVar component. It's basically configuration and as a bonus is it'll also make it easier to test.